esp-hal/src/rmt.rs

```diff
@@ -2418,3 +2418,128 @@ mod chip_specific {
+        ) -> Option<Self> {
+            if target > source {
+                return None;
+            }
+
+            let quotient = source / target;
+            if quotient > integral_max {
+                return None;
+            }
+
+            let remainder = source % target;
+            if remainder == 0 {
+                return Some(Self {
+                    div_num: quotient,
+                    div_b: 0,
+                    div_a: 0,
+                });
+            }
+
+            let gcd = gcd(target, remainder);
+            let numerator = remainder / gcd;
+            let denominator = target / gcd;
+            if numerator <= fractional_max && denominator <= fractional_max {
+                return Some(Self {
+                    div_num: quotient,
+                    div_b: numerator,
+                    div_a: denominator,
+                });
+            }
+
+            // Search Stern-Brocot Tree
+            let target_frac = Fraction {
+                numerator,
+                denominator,
+            };
+            let mut l = Fraction {
+                numerator: 0,
+                denominator: 1,
+            };
+            let mut h = Fraction {
+                numerator: 1,
+                denominator: 1, // We only search the left harf part
+            };
+            loop {
+                let m = Fraction {
+                    numerator: l.numerator + h.numerator,
+                    denominator: l.denominator + h.denominator,
+                };
+                if m.numerator > fractional_max || m.denominator > fractional_max {
+                    // L and H, which is closer?
+
+                    // M - L
+                    let a = Fraction {
+                        numerator: m.numerator * l.denominator - m.denominator * l.numerator,
+                        denominator: m.denominator * l.denominator,
+                    };
+                    // H - M
+                    let b = Fraction {
+                        numerator: h.numerator * m.denominator - h.denominator * m.numerator,
+                        denominator: h.denominator * m.denominator,
+                    };
+
+                    // compare a and b
+                    let c = a.numerator * b.denominator;
+                    let d = a.denominator * b.numerator;
+                    let m = if c < d { l } else { h };
+                    return Some(Self {
+                        div_num: quotient,
+                        div_b: m.numerator,
+                        div_a: m.denominator,
+                    });
+                }
+
+                // compare M and target_frac
+                let a = m.numerator * target_frac.denominator;
+                let b = m.denominator * target_frac.numerator;
+                if a > b {
+                    h = m;
+                } else if a < b {
+                    l = m;
+                } else {
+                    // unreachable!();
```

Comment:
left-over from testing?

Reply:
This is unreachable, but `unreachable!()` fails to built. So I just comment it out, and write some handling code technically correct.

Or we can use `unsafe unreachable_unchecked()`.